swarm/api/http/server.go

```diff
@@ -350,14 +462,24 @@ func (s *Server) HandleGet(w http.ResponseWriter, r *Request) {
 
 	switch {
 	case r.uri.Raw():
+		m := &api.Manifest{}
+		sz, _ := reader.Size(nil)
+		b := make([]byte, sz)
+		reader.Read(b)
+		err = json.Unmarshal(b, m)
+		if len(m.Entries) > 0 {
+			if m.Entries[0].ContentType == api.ResourceContentType {
```

Comment:
So the server tries to parse every raw request as a manifest? Should it not only do so if the URI is for a resource?

Reply:
@zelig cleared up yesterday that the intention behind manifests in the [https://github.com/ethersphere/go-ethereum/issues/180](outstanding issues list) was to transparently serve content from mutable resources given a certain content type.

Reply:
I'm not sure I understand. So you're saying if I store a really large file in Swarm and then download it via `bzz-raw`, then the server is going to first read the data into memory and try and parse it as a JSON manifest, just to check if it is a resource update?

Reply:
No. It shouldn't be like that. This code should be on the `bzz` scheme (which is what expects a manifest, right?). In any case not all should be loaded into memory, only enough bytes to determine if it's a manifest and if it's the right content-type.

Reply:
precisely